models/demos/t3000/llama2_70b/demo/eval.py

```diff
@@ -14,34 +15,57 @@
 from loguru import logger
 
 from tqdm import tqdm
-from models.demos.t3000.llama2_70b.demo.demo import run_decode, build_generator
+from models.demos.t3000.llama2_70b.demo.demo import (
+    run_decode,
+    build_generator,
+    construct_arg,
+    initialize_inputs,
+    get_sampling_func,
+)
 from datetime import datetime
 from models.demos.t3000.llama2_70b.tt.llama_common import (
     setup_llama_env,
     check_device_mesh,
 )
 
 
-def main(args):
+@dataclass
+class EvalDataArgs:
+    dataset: str = "wikitext"
+    split: str = "test"
+    config: str = "wikitext-2-raw-v1"
+    stride: int = 128
+    sample_len: int = 128
+    num_samples: int = 32
```

Comment:
ideally, `num_samples` should be some multiple of the current batch size? because iirc, we don't support in betweens. also if we force it to be a multiple of current batch size, we don't have to add checks for batch size 16 vs 32

Reply:
Ideally its 128 by default, since 32 gives consistent lower values